### PR TITLE
Add terra-consumer support

### PR DESCRIPTION
**Pull request by mhemesath**
### Summary
* Adds consumer support
* Switches CSS/JSX output to use CSS modules

Addresses issue #61

Thanks for contributing to Terra. 
@cerner/terra

[CONTRIBUTORS.md]: ../blob/master/CONTRIBUTORS.md
[License]: ../blob/master/LICENSE

### CODE REVIEW

File: test/app.js [hunk 1]
```diff
@@ -80,78 +83,88 @@ describe('generator-terra-module:app', function () {
...
-      assert.fileContent('packages/terra-clinical-site/src/examples/monster-cookies/Index.jsx', `import { version } from 'terra-clinical-monster-cookies/package.json';`);
-      assert.fileContent('packages/terra-clinical-site/src/examples/monster-cookies/Index.jsx', `export default MonsterCookiesExamples;`);
-    });
-
-    it('fills the examples Index file with project data', function () {
-      assert.fileContent('packages/terra-clinical-monster-cookies/.npmignore', `src`);
-      assert.fileContent('packages/terra-clinical-monster-cookies/.npmignore', `node_modules`);
-      assert.fileContent('packages/terra-clinical-monster-cookies/.npmignore', `*.log`);
+  ['terra-clinical', 'terra-consumer'].forEach(repository => {
+    describe(repository, function () {
```
Comment (calebmeyer) [inline test/app.js]: Why use arrow functions for part of this file and old-school functions for other parts?

File: generators/app/templates/terra-consumer-package.json [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  "version": "0.0.0",
+  "description": "{insert description}",
+  "repository": {
+    "type": "git",
+    "url": "git+https://github.com/cerner/terra-consumer.git"
+  },
+  "keywords": [
+    "Cerner",
+    "Terra",
+    "Clinical",
```
Comment (Matt-Butler) [inline generators/app/templates/terra-consumer-package.json]: Consumer

File: test/app.js (hunk 1 above)
Comment (mhemesath) [inline test/app.js]: the whole file should be updated. Quite honestly the tests should be cleaned up a bit and simplified... this should be done in a separate pull request IMO.

File: generators/app/index.js [hunk 3]
```diff
@@ -93,7 +84,7 @@ module.exports = yeoman.Base.extend({
       this.props.projectClassName = toClassName(this.props.projectName);
       this.props.titlecaseProjectName = toTitleCase(this.props.projectName.replace('-', ' '));
 
-      this.props.cssClassName = toCssClassName(this.props.repoPrefix, this.props.moduleName);
+      this.props.cssClassName = _.camelCase(this.props.moduleName);
```
Comment (emilyrohrbough) [inline generators/app/index.js]: Is camelCase the format we want to follow? In terra-clinical css names have been using the format `aaa-bbb`

File: generators/app/templates/terra-consumer-package.json [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    "test:nightwatch-default": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",
+    "test:nightwatch-chrome": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js chrome",
+    "test:nightwatch-firefox": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js firefox",
+    "test:nightwatch-safari": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-non-parallel.js safari",
+    "test:remote": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> REMOTE=true node ../../node_modules/terra-toolkit/lib/scripts/nightwatch-process.js --config tests/nightwatch.conf.js",
```
Comment (emilyrohrbough) [inline generators/app/templates/terra-consumer-package.json]: Remove remote test scripts.

File: generators/app/templates/terra-consumer-package.json [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+    "compile:clean": "$(cd ..; npm bin)/rimraf lib",
+    "compile:build": "$(cd ..; npm bin)/babel src --out-dir lib --copy-files",
+    "lint": "npm run lint:js && npm run lint:scss",
+    "lint:js": "$(cd ..; npm bin)/eslint --ext .js,.jsx . --ignore-path ../../.eslintignore",
+    "lint:scss": "$(cd ..; npm bin)/stylelint src/**/*.scss",
+    "props-table": "$(cd ..; npm bin)/props-table ./src/<%=jsxFileName%>.jsx --out-dir ./docs/props-table",
+    "test": "npm run test:spec && npm run test:nightwatch-default",
+    "test:spec": "$(cd ..; npm bin)/jest --config ../../jestconfig.json",
+    "test:all": "npm run test:nightwatch-default && npm run test:nightwatch-chrome && npm run test:nightwatch-firefox && npm run test:nightwatch-safari",
+    "test:nightwatch-default": "WEBPACK_CONFIG_PATH=../../../../terra-consumer-site/webpack.config SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",
```
Comment (emilyrohrbough) [inline generators/app/templates/terra-consumer-package.json]: These can be updated to:

`"test:nightwatch-default": " SPECTRE_TEST_SUITE=<%= projectName %> node ../../node_modules/terra-toolkit/lib/scripts/nightwatch.js",`

Comment (emilyrohrbough) [inline generators/app/templates/terra-consumer-package.json]: No longer need the `WEBPACK_CONFIG_PATH `with the toolkit update--that is if the webpack config being used by terra-consumer-site is the same as terra-core and terra-clinical.

File: generators/app/templates/tests/terra-consumer-nightwatch.conf.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+// eslint-disable-next-line import/no-extraneous-dependencies
```
Comment (emilyrohrbough) [inline generators/app/templates/tests/terra-consumer-nightwatch.conf.js]: If using the same webpack config as terra-core and terra-clinical, this file is not longer needed.

File: test/app.js (hunk 1 above)
Comment (emilyrohrbough) [inline test/app.js]: Agreed. At this point, the only differences between the repositories would be the template package file-the differences being the props table script and the key words used.

File: test/app.js [hunk 7]
```diff
@@ -3,6 +3,8 @@ var path = require('path');
 var assert = require('yeoman-assert');
 var helpers = require('yeoman-test');
 
+const titleize = word => `${word.charAt(0).toUpperCase()}${word.slice(1)}`;
```
Comment (emilyrohrbough) [inline test/app.js]: What does word stand for? I see its used in the mapping below but a comment might be helpful for quick reference.
